pytorch_lightning/strategies/deepspeed.py

```diff
@@ -68,25 +68,28 @@ def remove_module_hooks(model: torch.nn.Module) -> None:
 
 class LightningDeepSpeedModule(_LightningModuleWrapperBase):
     def __init__(
-        self, pl_module: Union["pl.LightningModule", _LightningPrecisionModuleWrapperBase], precision: int
+        self, pl_module: Union["pl.LightningModule", _LightningPrecisionModuleWrapperBase], precision: Union[str, int]
     ) -> None:
         super().__init__(pl_module)
         self.precision = precision
 
     def forward(self, *inputs, **kwargs):
-        if self.precision == 16:
-            inputs = self._move_float_tensors_to_half(inputs)
-
+        inputs = self._move_float_tensors_to_dtype(inputs, function=self._batch_to)
```

Comment:
Let's remove the staticmethod

```suggestion
        inputs = apply_to_collection(batch, torch.Tensor, function=self._batch_to)
```